src/viam/sdk/common/proto_type.hpp

```diff
@@ -8,6 +8,9 @@
 namespace viam {
 namespace sdk {
 
+class ProtoType;
+using AttributeMap = std::shared_ptr<std::unordered_map<std::string, std::shared_ptr<ProtoType>>>;
```

Comment:
Drive-by. This typedef is more useful in this header than in `resource.hpp` and moving it here simplified things a bit. There is more work needed on `ProtoType`, some of which is in #101. I've resisted the urge to make those changes here in favor of waiting on that PR.